plugins/basic/nexus-repository-docker/src/main/java/org/sonatype/nexus/repository/docker/internal/DockerConnectorFacetImpl.java

```diff
@@ -0,0 +1,281 @@
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.repository.docker.internal;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.validation.ConstraintViolation;
+
+import org.sonatype.nexus.bootstrap.jetty.ConnectorConfiguration;
+import org.sonatype.nexus.bootstrap.jetty.ConnectorRegistrar;
+import org.sonatype.nexus.repository.FacetSupport;
+import org.sonatype.nexus.repository.Repository;
+import org.sonatype.nexus.repository.config.Configuration;
+import org.sonatype.nexus.repository.config.ConfigurationFacet;
+import org.sonatype.nexus.repository.docker.DockerConnectorFacet;
+import org.sonatype.nexus.repository.manager.RepositoryManager;
+import org.sonatype.nexus.repository.storage.StorageFacet;
+import org.sonatype.nexus.validation.ConstraintViolationFactory;
+import org.sonatype.nexus.validation.ConstraintViolations;
+import org.sonatype.nexus.validation.constraint.PortNumber;
+
+import org.eclipse.jetty.http.HttpScheme;
+import org.eclipse.jetty.http.HttpURI;
+import org.eclipse.jetty.server.Connector;
+import org.eclipse.jetty.server.HttpConfiguration;
+import org.eclipse.jetty.server.HttpConfiguration.Customizer;
+import org.eclipse.jetty.server.Request;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * @since 3.0
+ */
+@Named
+public class DockerConnectorFacetImpl
+    extends FacetSupport
+    implements DockerConnectorFacet
+{
+  static final String CONFIG_KEY = "docker";
+
+  private final RepositoryManager repositoryManager;
+
+  private final ConstraintViolationFactory constraintViolationFactory;
+
+  private final ConnectorRegistrar connectorRegistrar;
+
+  private Config config;
+
+  private ConnectorConfiguration httpConfig;
+
+  private ConnectorConfiguration httpsConfig;
+
+  @Inject
+  public DockerConnectorFacetImpl(final RepositoryManager repositoryManager,
+                                  final ConstraintViolationFactory constraintViolationFactory,
+                                  final ConnectorRegistrar connectorRegistrar)
+  {
+    this.repositoryManager = checkNotNull(repositoryManager);
+    this.constraintViolationFactory = checkNotNull(constraintViolationFactory);
+    this.connectorRegistrar = checkNotNull(connectorRegistrar);
+  }
+
+  @Override
+  protected void doValidate(final Configuration configuration) throws Exception {
+    facet(ConfigurationFacet.class).validateSection(configuration, CONFIG_KEY, Config.class);
+
+    Config configToValidate = facet(ConfigurationFacet.class).readSection(configuration, CONFIG_KEY, Config.class);
+    Set<ConstraintViolation<?>> violations = new HashSet<>();
+    maybeAdd(violations, validatePortNotInUse(pathOf("httpPort"), configToValidate.httpPort));
+    maybeAdd(violations, validatePortNotInUse(pathOf("httpsPort"), configToValidate.httpsPort));
```

Comment:
Will not do that as that will prevent the repository form being created for example on a restart. 
For now if connectors fail to be created, the failure will be logged but repository will be in working state.